crates/bevy_ecs/src/component.rs

```diff
@@ -2126,42 +2142,80 @@ pub fn component_clone_via_clone<C: Clone + Component>(
 ///
 /// See [`ComponentCloneHandlers`] for more details.
 #[cfg(feature = "bevy_reflect")]
-pub fn component_clone_via_reflect(world: &mut DeferredWorld, entity_cloner: &EntityCloner) {
-    let component_id = entity_cloner.component_id();
-    let source = entity_cloner.source();
-    let target = entity_cloner.target();
-    world.commands().queue(move |world: &mut World| {
-        world.resource_scope::<crate::reflect::AppTypeRegistry, ()>(|world, registry| {
+pub fn component_clone_via_reflect(world: &mut DeferredWorld, ctx: &mut ComponentCloneCtx) {
+    // This path will be selected if:
+    // 1. Component has ReflectFromReflect or ReflectDefault registered.
+    // 2. Component has ReflectFromPtr registered.
+    // Otherwise, it will fallback to clone_slow.
+    fn clone_fast(ctx: &mut ComponentCloneCtx) -> Option<()> {
+        let registry = ctx.type_registry()?;
+        let component = {
+            let source_component_reflect = ctx.read_source_component_reflect()?;
             let registry = registry.read();
+            let component_info = ctx.components().get_info(ctx.component_id())?;
+            let type_id = component_info.type_id()?;
+
+            if let Some(reflect_from_reflect) =
+                registry.get_type_data::<bevy_reflect::ReflectFromReflect>(type_id)
+            {
+                reflect_from_reflect.from_reflect(source_component_reflect.as_partial_reflect())
+            } else if let Some(reflect_default) =
+                registry.get_type_data::<bevy_reflect::std_traits::ReflectDefault>(type_id)
```

Comment:
We can also try `ReflectFromWorld` like `ReflectComponent` does (though it's not as commonly registered).

Reply:
Wouldn't that require us to give access to `&mut World` to clone handlers (which will be unsound) then?

Reply:
Could we not queue it up as a command like you have for `clone_slow`?